### ISSUE TITLE
Malformed DR lines in RepBase EMBL files break parser

### ISSUE THREAD

**Issue by juancrescente** [biopython/biopython#1579]
### Setup

biopython==1.70
Python 2.7.12 

### Description

Reading embl from repbase:

http://www.girinst.org/server/RepBase/protected/RepBase23.02.embl.tar.gz

Manually checked the file and seems fine.

```python
fasta_seq = SeqIO.parse(args.embl, 'embl')
for record in fasta_seq:
   pass
```

```
  File "/usr/local/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 676, in read
    first = next(iterator)
  File "/usr/local/lib/python2.7/site-packages/Bio/SeqIO/__init__.py", line 611, in parse
    for r in i:
  File "/usr/local/lib/python2.7/site-packages/Bio/GenBank/Scanner.py", line 486, in parse_records
    record = self.parse(handle, do_features)
  File "/usr/local/lib/python2.7/site-packages/Bio/GenBank/Scanner.py", line 470, in parse
    if self.feed(handle, consumer, do_features):
  File "/usr/local/lib/python2.7/site-packages/Bio/GenBank/Scanner.py", line 438, in feed
    self._feed_header_lines(consumer, self.parse_header())
  File "/usr/local/lib/python2.7/site-packages/Bio/GenBank/Scanner.py", line 869, in _feed_header_lines
    parts[1].strip()))
IndexError: list index out of range
```
### Expected behaviour

No error

**Comment by peterjc** [comment]
We've seen invalid data from RepBase before, e.g. #910

I can't access that file - it wants a username/password. Can you extract a complete failing record from the file, and share that instead? I am not a lawyer, but think that should be fine from a fair use point of view.

**Comment by juancrescente** [comment]
This is a problematic record
`
ID   MuDR-N273_OS repbase;    DNA;    ORY; 799 BP.
XX
AC   .
XX
DT   26-OCT-2017 (Rel. 22.11, Created)
DT   26-OCT-2017 (Rel. 22.11, Last updated, Version 1)
XX
DE   DNA transposon - consensus.
XX
KW   MuDR; DNA transposon; Transposable Element; Nonautonomous;
KW   MuDR-N273_OS.
XX
OS   Oryza sativa
OC   Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;
OC   Spermatophyta; Magnoliophyta; Liliopsida; Poales; Poaceae;
OC   BOP clade; Oryzoideae; Oryzeae; Oryzinae; Oryza.
XX
RN   [1]
RP   1-799
RA   Bao W.;
RT   "DNA transposons from the rice genome.";
RL   Repbase Reports 17(11), 1316-1316 (2017).
XX
DR   [1] (Consensus)
XX
CC   ~91% identical to the consensus.
XX
SQ   Sequence 799 BP; 262 A; 122 C; 125 G; 289 T; 1 other;
     ggggtttttt tttaactatt tgccactctt acggagtgsc actattgaga tgccactctt        60
     ccaaattttt ttacacaaac accacatgag agagatattt ttcctaacta gagtgccatg       120
     tcagcaagag agacttgaaa tgaccttttt acccctagtt tgagaaatat gagagattat       180
     atacaaacat atataagaaa atattattta ttatatctta aaaagagtta tcacgggcaa       240
     atatacgtcg aacaagttca ccataattta ataatagatt caacataggg tgaacttgtt       300
     caacatatat ttatccgtga taactctttt tagatataat atagtacgta cttactatgt       360
     tgaatctatt attaaattgt ggtgaacttg tccaacgtat actgctgtga taactctttt       420
     ttagatacaa tatattgtgt tgaatctatt attaaattgt ggtgaacttg ttcaacgtat       480
     attgcccgtg ataaattttt ttagatataa tatactatgc taaatctatt attaaattgt       540
     ggtgaacttg ttcaacgtat atttgcccgt gataactctt ttttaagata taataaacaa       600
     tattttctta tatatgtttg tatataatct ctcatatttc tcaaactagg ggtaaaaagg       660
     tcatttcaag tcccccttgc tgacatggca ctctagttag gaaaaatatc tctctcatgt       720
     ggtgtttgtg taaaaaaatt tggaagagtg gcatctcaat agtggcactc cgtaagagtg       780
     gcaaatagtt aaaaacccc                                                    799
`

**Comment by juancrescente** [comment]
This is a problematic record:


ID   MuDR-N273_OS repbase;    DNA;    ORY; 799 BP.
XX
AC   .
XX
DT   26-OCT-2017 (Rel. 22.11, Created)
DT   26-OCT-2017 (Rel. 22.11, Last updated, Version 1)
XX
DE   DNA transposon - consensus.
XX
KW   MuDR; DNA transposon; Transposable Element; Nonautonomous;
KW   MuDR-N273_OS.
XX
OS   Oryza sativa
OC   Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;
OC   Spermatophyta; Magnoliophyta; Liliopsida; Poales; Poaceae;
OC   BOP clade; Oryzoideae; Oryzeae; Oryzinae; Oryza.
XX
RN   [1]
RP   1-799
RA   Bao W.;
RT   "DNA transposons from the rice genome.";
RL   Repbase Reports 17(11), 1316-1316 (2017).
XX
DR   [1] (Consensus)
XX
CC   ~91% identical to the consensus.
XX
SQ   Sequence 799 BP; 262 A; 122 C; 125 G; 289 T; 1 other;
     ggggtttttt tttaactatt tgccactctt acggagtgsc actattgaga tgccactctt        60
     ccaaattttt ttacacaaac accacatgag agagatattt ttcctaacta gagtgccatg       120
     tcagcaagag agacttgaaa tgaccttttt acccctagtt tgagaaatat gagagattat       180
     atacaaacat atataagaaa atattattta ttatatctta aaaagagtta tcacgggcaa       240
     atatacgtcg aacaagttca ccataattta ataatagatt caacataggg tgaacttgtt       300
     caacatatat ttatccgtga taactctttt tagatataat atagtacgta cttactatgt       360
     tgaatctatt attaaattgt ggtgaacttg tccaacgtat actgctgtga taactctttt       420
     ttagatacaa tatattgtgt tgaatctatt attaaattgt ggtgaacttg ttcaacgtat       480
     attgcccgtg ataaattttt ttagatataa tatactatgc taaatctatt attaaattgt       540
     ggtgaacttg ttcaacgtat atttgcccgt gataactctt ttttaagata taataaacaa       600
     tattttctta tatatgtttg tatataatct ctcatatttc tcaaactagg ggtaaaaagg       660
     tcatttcaag tcccccttgc tgacatggca ctctagttag gaaaaatatc tctctcatgt       720
     ggtgtttgtg taaaaaaatt tggaagagtg gcatctcaat agtggcactc cgtaagagtg       780
     gcaaatagtt aaaaacccc                                                    799

**Comment by juancrescente** [comment]
> ID   MuDR-N273_OS repbase;    DNA;    ORY; 799 BP.
> XX
> AC   .
> XX
> DT   26-OCT-2017 (Rel. 22.11, Created)
> DT   26-OCT-2017 (Rel. 22.11, Last updated, Version 1)
> XX
> DE   DNA transposon - consensus.
> XX
> KW   MuDR; DNA transposon; Transposable Element; Nonautonomous;
> KW   MuDR-N273_OS.
> XX
> OS   Oryza sativa
> OC   Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;
> OC   Spermatophyta; Magnoliophyta; Liliopsida; Poales; Poaceae;
> OC   BOP clade; Oryzoideae; Oryzeae; Oryzinae; Oryza.
> XX
> RN   [1]
> RP   1-799
> RA   Bao W.;
> RT   "DNA transposons from the rice genome.";
> RL   Repbase Reports 17(11), 1316-1316 (2017).
> XX
> DR   [1] (Consensus)
> XX
> CC   ~91% identical to the consensus.
> XX
> SQ   Sequence 799 BP; 262 A; 122 C; 125 G; 289 T; 1 other;
>      ggggtttttt tttaactatt tgccactctt acggagtgsc actattgaga tgccactctt        60
>      ccaaattttt ttacacaaac accacatgag agagatattt ttcctaacta gagtgccatg       120
>      tcagcaagag agacttgaaa tgaccttttt acccctagtt tgagaaatat gagagattat       180
>      atacaaacat atataagaaa atattattta ttatatctta aaaagagtta tcacgggcaa       240
>      atatacgtcg aacaagttca ccataattta ataatagatt caacataggg tgaacttgtt       300
>      caacatatat ttatccgtga taactctttt tagatataat atagtacgta cttactatgt       360
>      tgaatctatt attaaattgt ggtgaacttg tccaacgtat actgctgtga taactctttt       420
>      ttagatacaa tatattgtgt tgaatctatt attaaattgt ggtgaacttg ttcaacgtat       480
>      attgcccgtg ataaattttt ttagatataa tatactatgc taaatctatt attaaattgt       540
>      ggtgaacttg ttcaacgtat atttgcccgt gataactctt ttttaagata taataaacaa       600
>      tattttctta tatatgtttg tatataatct ctcatatttc tcaaactagg ggtaaaaagg       660
>      tcatttcaag tcccccttgc tgacatggca ctctagttag gaaaaatatc tctctcatgt       720
>      ggtgtttgtg taaaaaaatt tggaagagtg gcatctcaat agtggcactc cgtaagagtg       780
>      gcaaatagtt aaaaacccc                                                    799
>

**Comment by juancrescente** [comment]
```
ID   MuDR-N273_OS repbase;    DNA;    ORY; 799 BP.
XX
AC   .
XX
DT   26-OCT-2017 (Rel. 22.11, Created)
DT   26-OCT-2017 (Rel. 22.11, Last updated, Version 1)
XX
DE   DNA transposon - consensus.
XX
KW   MuDR; DNA transposon; Transposable Element; Nonautonomous;
KW   MuDR-N273_OS.
XX
OS   Oryza sativa
OC   Eukaryota; Viridiplantae; Streptophyta; Embryophyta; Tracheophyta;
OC   Spermatophyta; Magnoliophyta; Liliopsida; Poales; Poaceae;
OC   BOP clade; Oryzoideae; Oryzeae; Oryzinae; Oryza.
XX
RN   [1]
RP   1-799
RA   Bao W.;
RT   "DNA transposons from the rice genome.";
RL   Repbase Reports 17(11), 1316-1316 (2017).
XX
DR   [1] (Consensus)
XX
CC   ~91% identical to the consensus.
XX
SQ   Sequence 799 BP; 262 A; 122 C; 125 G; 289 T; 1 other;
     ggggtttttt tttaactatt tgccactctt acggagtgsc actattgaga tgccactctt        60
     ccaaattttt ttacacaaac accacatgag agagatattt ttcctaacta gagtgccatg       120
     tcagcaagag agacttgaaa tgaccttttt acccctagtt tgagaaatat gagagattat       180
     atacaaacat atataagaaa atattattta ttatatctta aaaagagtta tcacgggcaa       240
     atatacgtcg aacaagttca ccataattta ataatagatt caacataggg tgaacttgtt       300
     caacatatat ttatccgtga taactctttt tagatataat atagtacgta cttactatgt       360
     tgaatctatt attaaattgt ggtgaacttg tccaacgtat actgctgtga taactctttt       420
     ttagatacaa tatattgtgt tgaatctatt attaaattgt ggtgaacttg ttcaacgtat       480
     attgcccgtg ataaattttt ttagatataa tatactatgc taaatctatt attaaattgt       540
     ggtgaacttg ttcaacgtat atttgcccgt gataactctt ttttaagata taataaacaa       600
     tattttctta tatatgtttg tatataatct ctcatatttc tcaaactagg ggtaaaaagg       660
     tcatttcaag tcccccttgc tgacatggca ctctagttag gaaaaatatc tctctcatgt       720
     ggtgtttgtg taaaaaaatt tggaagagtg gcatctcaat agtggcactc cgtaagagtg       780
     gcaaatagtt aaaaacccc                                                    799

```

**Comment by peterjc** [comment]
That seems to be missing the final line(s), is there not a ``\\`` too? You can edit the comment on GitHub to fix this.

**Comment by juancrescente** [comment]
Yes, I've added it

**Comment by peterjc** [comment]
It looks like the ``DR`` line is the problem,

```
DR   [1] (Consensus)
```

That is very odd. It looks more like an ``RN`` line. Could you run this on the full file:

```
$ grep ^DR RepBase23.02.embl | head
```

(Try without the head command too so see if there is anything else weird in there)

**Comment by peterjc** [comment]
Hmm, so lots of ``DR [1] (Consensus)`` cases and at least one more normal looking DR line.

I think we could add a hack to ignore ``DR [1] (Consensus)`` as a special case, but it would be best if you could contact RepBase first? Could you do that? Thanks!

**Comment by juancrescente** [comment]
Yes, what is the expected format for DR line?

**Comment by peterjc** [comment]
Quoting ftp://ftp.ebi.ac.uk/pub/databases/ena/sequence/release/doc/usrman.txt

```
3.4.11  The DR Line
The DR (Database Cross-reference) line cross-references other databases which
contain information related to the entry in which the DR line appears. For
example, if an annotated/assembled sequence in ENA is cited in the IMGT/LIGM
database there will be a DR line pointing to the relevant IMGT/LIGM entry.
The format of the DR line is as follows:
     DR   database_identifier; primary_identifier; secondary_identifier.
The first item on the DR line, the database identifier, is the abbreviated 
name of the data collection to which reference is made.
The second item on the DR line, the primary identifier, is a pointer to 
the entry in the external database to which reference is being made.
The third item on the DR line is the secondary identifier, if available, from
the referenced database.
An example of a DR line is shown below:
DR   MGI; 98599; Tcrb-V4.
```

**Comment by juancrescente** [comment]
I'll send to repbase but I'm not sure I'll get a response since I've already commented some issues in there

**Comment by peterjc** [comment]
Separately does this hack let you parse the whole file, or are there more issues:
https://github.com/peterjc/biopython/tree/repbase_issue1597

Please ask if you need help trying that change locally.

**Comment by OscarMaestre** [comment]
I also think that the file doesn't adhere to the standard format. However, I made a small hack to allow the file to be processed. I also created an associated test but I need the wrong file to be included in BioPython. Should I submit my patch?

**Comment by peterjc** [comment]
You are welcome to make a pull request with your hack. Depending on the approach, I might merge it anyway (although we agree getting RepBase to fix their data is the best outcome).

Did my hack from last week not work? https://github.com/peterjc/biopython/tree/repbase_issue1597

**Comment by OscarMaestre** [comment]
To be honest, I didn't check your patch (my shame). I created the pull request

**Comment by peterjc** [comment]
Your change is in the same area of the code - this indirectly tells me that it was only the DR lines which were causing problems in the current RepBase files.

**Comment by peterjc** [comment]
Closed via #1586, thank you.

**Comment by peterjc** [comment]
Looks like I didn't click the close button in April, trying again...
